packages/provider/src/RelayClient.ts

```diff
@@ -567,7 +584,8 @@ export class RelayClient {
         version: ''
       }
     }
-    const relayRequest = await this._prepareRelayRequest(gsnTransactionDetails, dryRunRelayInfo)
+    // TODO: clone?
+    this.fillRelayInfo(relayRequest, dryRunRelayInfo)
```

Comment:
why do you need to fill from a dummy relayInfo ? (and the callDatacost is also irrelevant, since it doesn't cover all fields.)